### DISCUSSION TITLE
✅ No-code "size:auto" for containers

### DISCUSSION THREAD

**Discussion by andreelb** [elementor/elementor#22960] [Feature Request]
### Prerequisites

- [X] I have searched for similar features requests in both open and closed tickets and cannot find a duplicate.
- [X] The feature is still missing in the latest stable version of Elementor ( Elementor Pro )

### What problem is your feature request going to solve? Please describe.

Let you shrink a container (especially with different background color) to the size of it's contents.

### Describe the solution you'd like

I'd like a built in option to shrink the size of a container to the size of its contents. Elementor likes to default to `width:100%` which isn't always ideal. Only other option is Boxed which limits the size of the contents, but not the size of the background.

Setting a % is the beginner way to do it but doesn't "break" nicely on font/resolution changes.

The setting would be under the Layout Tab, maybe in Content Width.

PS: I struggled with an erroneous understanding of flex shrink before I came across the correct custom css solution.

### Describe alternatives you've considered

Currently it can be done with custom CSS but would be nice to have it code free.
`selector {width:auto;}`

### Additional context

Example use case:

Intended result:
A floating header, with a separate background color, that adjusts to the size of the menu elements.

<img width="996" alt="image" src="https://github.com/elementor/elementor/assets/138502530/9318789b-5bde-48b6-94e3-050cc6a081fa">

Without the `width:auto` command, it occupies the full width and has extra empty space which can't be reduced in any code-free way. 

<img width="1138" alt="image" src="https://github.com/elementor/elementor/assets/138502530/8b7a50a4-4e18-436e-83de-0d9302c3a05b">

**Comment by nicholaszein** [comment]
Hello @andreelb!

Thank you for sharing your idea! 🙏 

We have great news though, the feature you're requesting is already possible by using [custom units](https://elementor.com/blog/new-mega-menu-loop-alternate-template-and-more/#:~:text=Choose%20any%20Unit%20Type%2C%20Anywhere). All you have to do is to set your Container's **`Content Width`** to **`Full Width`**, and change the unit of the **`Width`** control to **`Custom`**. Then just type `auto` and it should be good to go. 😉 

I'm closing this request as Solved.

Kind regards
